vortex-expr/src/literal.rs

```diff
@@ -17,6 +18,10 @@ impl Literal {
         Self { value }
     }
 
+    pub fn new_ref(value: Scalar) -> ExprRef {
```

Comment:
Agree with the type alias, not so sure about these constructor functions

Reply:
You could do `pub fn arced(self) -> ExprRef`

Reply:
`Thing::new(...).arced()` seems to me less legible than `Thing::new_arc` (or `_ref`) because it moves the words to the end of a possibly large expression.

Nick, can you elaborate on the ick?

Reply:
It seems very redundant from composition point of view. You end up repeating the new method just for the sake of wrapping it in an Arc. But... not quite sure how to name it and only ever alternative I have is the method I mentioned

Reply:
How about we just make new return a ref? Almost never will you want a stack allocated one

Reply:
Clippy hates that. "new should return Self"

Reply:
Though I agree with your second sentence.

Reply:
that's a fun clippy rule. Ok @gatesn do you have suggestions?

Reply:
`new_expr`?